packages/eslint-plugin-internal/src/rules/all-bases.js

```diff
@@ -17,6 +17,22 @@ module.exports = {
   // Enforce a maximum number of classes per file
   'max-classes-per-file': 'off',
 
+  // Disallow reassigning function parameters
+  'no-param-reassign': [
```

Comment:
@vojtechszocs my personal opinion would be to leave these rules configured as eslint has them by default:
* `no-param-reassign`: I feel React is heavily leaning towards `props` being treated as immutable. I feel turning this off would cause several anti-patterns. https://eslint.org/docs/latest/rules/no-param-reassign
* `no-plusplus`: I think we should favor loops that avoid the need for us to manage our own iterator unless it is explicitly required. I see in some of the tests below we use the ++ to populate an empty array. We could achieve that using something like `[...Array(20)].map((item, i) => { return { foo: i }});`. https://eslint.org/docs/latest/rules/no-plusplus

Reply:
Based on our discussion, I've reverted the custom configs for the above rules, as well as updating the existing `for` loops to use `index += 1` instead of `index++`.